### ISSUE TITLE
System Clipboard (paste loses data)

### ISSUE THREAD

**Issue by naps62** [neovim/neovim#3377]
I'm having some trouble pasting a 3000 line JSON string from the system clipboard into nvim.

I've always had problems integrating my system clipboard with vim (I'm using Arch Linux), but this issue seems to be present only in neovim, and not default vim

I'm doing what I usually do to use the system clipboard:

```
:set paste
<Shift-Insert>
:set nopaste
```

Using `/usr/bin/vim -u NONE` (the original vim, not neovim) this works correctly, only with a delay of about 4-5 seconds.

Doing the same thing with `nvim -u NONE`, it hangs for a lot longer (about 10 seconds i'd say), and the result is just about 800 lines long, and with the formatting completely ruined: Here's a screenshot with vim on the left, and neovim on the right:

![screenshot from 2015-09-23 12-12-52](https://cloud.githubusercontent.com/assets/283819/10043622/72a3c3b4-61ec-11e5-8992-ce00262644d4.png)

This is running on UXterm + Tmux. But running outside tmux and with other terminal emulators seems to have the same result

Doing `xclip -selection clipboard -o | nvim -` also works correctly, so this seems to be an issue with nvim.

I'm running the latest version, compiled locally from the master branch:
```
NVIM 0.0.0-alpha+201509230028 (compiled Sep 23 2015 11:57:03)
Commit: ba6c7a6f5b45ce5d79e6dc4838a967b2299b524c
```

**Comment by justinmk** [comment]
See https://github.com/neovim/neovim/issues/1714. Thanks

**Comment by tarruda** [comment]
@justinmk This is actually another problem which I'm fixing in #3360

**Comment by justinmk** [comment]
Ok, would be great if this could be covered by a test(s) so we never have to worry about this again :)

**Comment by tarruda** [comment]
> Ok, would be great if this could be covered by a test(s) so we never have to worry about this again :)

I have a few ideas on how to automate TUI tests, but it requires some time to put together. One simple way is to reuse our lua screen testing infrastructure, and spawn another nvim instance in a `:terminal` window, which we could send input and verify output like most other screen tests.

**Comment by tarruda** [comment]
Should be closed by #3360
